src/pcapi/scripts/payment/batch_steps.py

```diff
@@ -1,86 +1,128 @@
+    ids = payments.with_entities(Payment.id)
+    payments = db.session.query(Payment).filter(Payment.id.in_(ids))
+    payments.update({"batchDate": batch_date}, synchronize_session=False)
+    db.session.commit()
+
+
+def get_venues_to_reimburse() -> Iterable[tuple[id, str]]:
+    return tuple(
+        db.session.execute(
+            """
+        SELECT id, CASE WHEN "publicName" IS NOT NULL THEN "publicName" ELSE name END
+        FROM venue
+        WHERE id IN (
+            SELECT DISTINCT(venue.id)
+            FROM venue
+            JOIN offer ON offer."venueId" = venue.id
+            JOIN stock ON stock."offerId" = offer.id
+            JOIN booking ON booking."stockId" = stock.id
+            LEFT OUTER JOIN payment ON payment."bookingId" = booking.id
+            WHERE
+                booking."isUsed" AND NOT booking."isCancelled"
+                AND payment.id IS NULL
         )
+            """
+        )
+    )
 
-        with db.session.no_autoflush:
-            payments = list(map(create_payment_for_booking, booking_reimbursements_to_pay))
 
-        if payments:
-            repository.save(*payments)
-            all_payments.extend(payments)
-        logger.info("[BATCH][PAYMENTS] Saved %i payments for offerer : %s", len(payments), offerer.name)
+def generate_new_payments(batch_date: datetime) -> None:
+    n_payments = 0
+    venues_to_reimburse = get_venues_to_reimburse()
+    for venue_id, venue_name in venues_to_reimburse:
+        bookings = booking_repository.find_bookings_eligible_for_payment_for_venue(venue_id)
+        reimbursements = find_all_booking_reimbursements(bookings, RULES)
+        to_pay = filter_out_already_paid_for_bookings(filter_out_bookings_without_cost(reimbursements))
+        if not to_pay:
+            logger.info("[BATCH][PAYMENTS] No payments generated for venue: %s", extra={"venue": venue_id})
+            continue
+        n_payments += len(to_pay)
+
+        payments = [create_payment_for_booking(reimbursement, batch_date) for reimbursement in to_pay]
+        db.session.bulk_save_objects(payments)
+        db.session.commit()
+
+        logger.info(
+            "[BATCH][PAYMENTS] Saved %i payments for venue: %s",
+            len(payments),
+            venue_name,
+            extra={"venue": venue_id, "payments": len(payments)},
+        )
 
-    logger.info("[BATCH][PAYMENTS] Generated %i payments for %i offerers in total", len(all_payments), len(offerers))
+    # Create all payment statuses. We get payments created above by
+    # looking at their transaction label. A bit ugly, but it works.
+    if n_payments:
+        # FIXME (dbaty, 2021-05-31): try to use `bulk_create_payment_statuses()` instead.
+        # In that function, we could try to use the ORM instead of raw
+        # SQL, and the function have an extra ``**filters` argument.
+        # We could then call it twice here with the propert status and
```

Comment:
```suggestion
        # We could then call it twice here with the proper status and
```
au cas où, mais je vais tenter de trouver une requête ORM qui aille aussi vite sans faire de raw sql :)

Reply:
vu ensemble : quand j'ai écrit ce FIXME, `bulk_create_payment_statuses()` utilisait du SQL brut. Maintenant ça utilise l'ORM et on pourrait facilement l'appeler. C'est ce que j'ai fait, finalement.